packages/geolocator/tizen/src/geolocator_tizen_plugin.cc

```diff
@@ -49,200 +143,126 @@ class GeolocatorTizenPlugin : public flutter::Plugin {
-        [this](const flutter::EncodableValue *arguments,
-               std::unique_ptr<flutter::EventSink<>> &&events)
-            -> std::unique_ptr<flutter::StreamHandlerError<>> {
-          OnListenGeolocatorUpdates(std::move(events));
-          return nullptr;
-        },
-        [this](const flutter::EncodableValue *arguments)
-            -> std::unique_ptr<flutter::StreamHandlerError<>> {
-          OnCancelGeolocatorUpdates();
-          return nullptr;
-        });
-    geolocator_updates_channel_->SetStreamHandler(std::move(handler));
-  }
+    result_ = std::move(result);
 
-  void HandleMethodCall(
-      const flutter::MethodCall<flutter::EncodableValue> &method_call,
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
-    std::string method_name = method_call.method_name();
     if (method_name == "checkPermission") {
-      OnCheckPermission(std::move(result));
+      OnCheckPermission();
     } else if (method_name == "isLocationServiceEnabled") {
-      OnIsLocationServiceEnabled(std::move(result));
+      OnIsLocationServiceEnabled();
     } else if (method_name == "requestPermission") {
-      OnRequestPermission(std::move(result));
+      OnRequestPermission();
     } else if (method_name == "getLastKnownPosition") {
-      OnGetLastKnownPosition(std::move(result));
+      OnGetLastKnownPosition();
     } else if (method_name == "getCurrentPosition") {
-      OnGetCurrentPosition(std::move(result));
+      OnGetCurrentPosition();
     } else if (method_name == "openAppSettings") {
-      TizenResult ret = Setting::LaunchAppSetting();
-      result->Success(flutter::EncodableValue(static_cast<bool>(ret)));
+      bool opened = app_settings_manager_->OpenAppSettings();
+      SendResult(flutter::EncodableValue(opened));
     } else if (method_name == "openLocationSettings") {
-      TizenResult ret = Setting::LaunchLocationSetting();
-      result->Success(flutter::EncodableValue(static_cast<bool>(ret)));
+      bool opened = app_settings_manager_->OpenLocationSettings();
+      SendResult(flutter::EncodableValue(opened));
     } else {
       result->NotImplemented();
     }
   }
 
-  void OnCheckPermission(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
-    PermissionStatus permission_status;
-    TizenResult ret =
-        permission_manager_->CheckPermissionStatus(&permission_status);
-    if (!ret) {
-      result->Error("Failed to check permssion status.", ret.message());
+  void OnCheckPermission() {
+    PermissionStatus result =
+        permission_manager_->CheckPermission(kPrivilegeLocation);
+    if (result == PermissionStatus::kError) {
+      SendErrorResult("Operation failed", "Permission request failed.");
       return;
     }
-    LOG_INFO("permission_status is %d", permission_status);
-    result->Success(
-        flutter::EncodableValue(static_cast<int>(permission_status)));
+    LOG_INFO("permission_status is %d", result);
```

Comment:
Could you remove this line or format the message properly?

Reply:
I removed it. it is unnecessary